superset-frontend/src/explore/components/controls/ConditionalFormattingControl/ConditionalFormattingControl.tsx

```diff
@@ -89,19 +84,15 @@ const ConditionalFormattingControl = ({
   }, [conditionalFormattingConfigs, onChange]);
 
   // remove formatter when corresponding column is removed from controls
-  const removeFormattersWhenColumnsChange = useCallback(() => {
-    const newFormattingConfigs = conditionalFormattingConfigs.filter(config =>
-      columnOptions.some(option => option?.value === config?.column),
-    );
-    if (
-      newFormattingConfigs.length !== conditionalFormattingConfigs.length &&
-      onChange
-    ) {
-      setConditionalFormattingConfigs(newFormattingConfigs);
-      onChange(newFormattingConfigs);
-    }
-  }, [JSON.stringify(columnOptions)]);
-  useComponentDidUpdate(removeFormattersWhenColumnsChange);
+  const newFormattingConfigs = conditionalFormattingConfigs.filter(config =>
```

Comment:
Can we wrap it in useEffect?

Reply:
Wrapped.
Can you please review my code again? 
Thank you!